pandas/core/internals/blocks.py

```diff
@@ -2050,12 +2051,15 @@ def get_values(self, dtype=None):
 class DatetimeBlock(DatetimeLikeBlockMixin, Block):
     __slots__ = ()
     is_datetime = True
-    _can_hold_na = True
 
     def __init__(self, values, placement, ndim=None):
         values = self._maybe_coerce_values(values)
         super().__init__(values, placement=placement, ndim=ndim)
 
+    @property
+    def _can_hold_na(self):
+        return True
+
```

Comment:
This threw this error because in this class `_can_hold_na` is a fixed boolean variable and in the other mixin it's a property method. I made this one a property method to match.

`pandas/core/internals/blocks.py:2200: error: Definition of "_can_hold_na" in base class "ExtensionBlock" is incompatible with definition in base class "DatetimeBlock"`

Reply:
Makes sense to me